### PR TITLE
EIP-681 support + Deeplinks e2e

### PR DESCRIPTION
**Pull request by brunobar79**
- Adds support for `ethereum:` urls via deeplinks, app scanner, and phone camera app
- includes e2e tests for all scenarios including L2s
- Includes e2e tests for rainbow.me universal links

To test it use: https://brunobar79.github.io/eip681-link-generator

### CODE REVIEW

File: src/screens/SendSheet.js [hunk 1]
```diff
@@ -347,6 +304,87 @@ export default function SendSheet(props) {
     [selected, sendUpdateAssetAmount, updateMaxInputBalance]
   );
 
+  // Update all fields passed via params if needed
+  useEffect(() => {
```
Comment (estebanmino) [inline src/screens/SendSheet.js:308]: do you usually put `useEffect`s in the middle of `useState`s?

File: src/screens/SendSheet.js [hunk 2]
```diff
@@ -347,6 +304,87 @@ export default function SendSheet(props) {
...
+    updateMaxInputBalance,
+  ]);
+
+  const [isValidAddress, setIsValidAddress] = useState(!!recipientOverride);
+  const [currentProvider, setCurrentProvider] = useState();
+  const { colors, isDarkMode } = useTheme();
+
+  const showEmptyState = !isValidAddress;
+  const showAssetList = isValidAddress && isEmpty(selected);
+  const showAssetForm = isValidAddress && !isEmpty(selected);
```
Comment (estebanmino) [inline src/screens/SendSheet.js:341]: I've seen we use `useMemo` for stuff similar to this

File: src/utils/ethereumUtils.js [hunk 3]
```diff
@@ -426,6 +445,78 @@ function openTransactionInBlockExplorer(hash, network) {
...
+    return;
+  }
+
+  const assetWithPrice = parseAssetsNative([asset], nativeCurrency)[0];
+
+  InteractionManager.runAfterInteractions(() => {
+    const params = { address, asset: assetWithPrice, nativeAmount };
+    if (isNativeStackAvailable || android) {
+      Navigation.handleAction(Routes.SEND_FLOW, {
+        params,
```
Comment (estebanmino) [inline src/utils/ethereumUtils.js:511]: I don't see we're using the network here, if the user gets a request for Optimism the asset would be ETH, would the user know / send the tx on optimism or mainnet?

**Review by skylarbarrera — APPROVED** [review]
🧯

File: src/utils/ethereumUtils.js (hunk 3 above)
Comment (brunobar79) [reply src/utils/ethereumUtils.js:511]: We get the network from the chainId, and then we request the asset for that network https://github.com/rainbow-me/rainbow/pull/2343/files#diff-34ca43434805a064f0600ef0ff5a7dbec13f0ae719ba1c19afeffe0fbd1d9b3fR459

File: src/screens/SendSheet.js (hunk 2 above)
Comment (brunobar79) [reply src/screens/SendSheet.js:341]: useMemo is a bit expensive to use too, so I try to use it whenever the logic it runs is worth memoizing. In this case isEmpty is not worth it.

File: src/screens/SendSheet.js (hunk 1 above)
Comment (brunobar79) [reply src/screens/SendSheet.js:331]: only when I code at 3AM lol. Just fixed it 😉

Comment (estebanmino) [reply src/screens/SendSheet.js:331]: fast af

**Review by estebanmino — APPROVED** [review]
👌

**Pull request merged**